modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/DistributedSqlConfiguration.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.query.h2;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.processors.configuration.distributed.DistributePropertyListener;
+import org.apache.ignite.internal.processors.configuration.distributed.DistributedConfigurationLifecycleListener;
+import org.apache.ignite.internal.processors.configuration.distributed.DistributedPropertyDispatcher;
+import org.apache.ignite.internal.processors.configuration.distributed.SimpleDistributedProperty;
+import org.apache.ignite.internal.processors.subscription.GridInternalSubscriptionProcessor;
+import org.apache.ignite.internal.util.future.GridFutureAdapter;
+
+import static org.apache.ignite.internal.cluster.DistributedConfigurationUtils.makeUpdateListener;
+import static org.apache.ignite.internal.cluster.DistributedConfigurationUtils.setDefaultValue;
+
+/**
+ * Distributed configuration of the indexing module.
+ */
+public class DistributedSqlConfiguration {
+    /** Property update message. */
+    private static final String PROPERTY_UPDATE_MESSAGE =
+        "SQL parameter '%s' was changed from '%s' to '%s'";
+
+    /** Default disabled SQL functions. */
+    public static final HashSet<String> DFLT_DISABLED_FUNCS = (HashSet<String>)Arrays.stream(new String[] {
+        "FILE_READ",
+        "FILE_WRITE",
+        "CSVWRITE",
+        "CSVREAD",
+        "MEMORY_FREE",
+        "MEMORY_USED",
+        "LOCK_MODE",
+        "LINK_SCHEMA",
+        "SESSION_ID",
+        "CANCEL_SESSION"
+    }).collect(Collectors.toSet());
+
+    /** */
+    private final IgniteLogger log;
+
+    /** Value of cluster time zone. */
+    private final SimpleDistributedProperty<HashSet<String>> disabledSqlFuncs
+        = new SimpleDistributedProperty<>("sql.disabledFunctions");
+
+    /**
+     * @param isp Subscription processor.
+     * @param ctx Kernal context.
+     */
+    public DistributedSqlConfiguration(
+        GridInternalSubscriptionProcessor isp,
+        GridKernalContext ctx,
```

Comment:
what reason to have context here?

Reply:
Good point. I guess log parameter must be removed and logger is created on constructo from ctx.log().

Fixed